codebase-pipelines/custom_pipeline_update/update_pipeline.py

```diff
@@ -0,0 +1,33 @@
+import os
+import json
+import boto3
+from typing import List
+
+
+def update_pipeline_stage_failure(pipelines: List[str]):
+    for pipeline_name in pipelines:
+        print(f"Updating pipeline {pipeline_name}")
+
+        client = boto3.client("codepipeline")
+
+        response = client.get_pipeline(name=pipeline_name)
+        pipeline = response["pipeline"]
+
+        stage_found = False
+        for stage in pipeline["stages"]:
+            if "Deploy" in stage["name"]:
+                stage["onFailure"] = {
+                    "result": "ROLLBACK"
+                }
+                stage_found = True
+
+        if not stage_found:
+            raise ValueError(f"Stage Deploy not found in pipeline {pipeline_name}")
+
+        client.update_pipeline(pipeline=pipeline)
```

Comment:
Could do with tests for:
-  sends in multiple pipelines and check that `update_pipeline` is called with JSON that has the onFailure set on the appropriate stages.
- asserts that `update_pipeline` is not called when an error condition is hit for a stage.